### ISSUE TITLE
Kazoo Test harness and real life testing integration

### ISSUE THREAD

**Issue by itissid** [python-zk/kazoo#133]
TL;DR Multiple py.test VM's running test harness cause issues as server instantiation is hardcoded. Change: https://gist.github.com/itissid/04d67590a6201b156fa8
shows that running separate customized test harnesses in parallel can works which is how a lot of testing integrations would want it to be .

The  harness for v 0.9 does not allow for multiple Test Harnesses to run with py.test instances to run (I don't believe the 1.3.1 test harness will work either). Multiple here means 2 Virtual Machines running independent tests with py.test. What seems to happen is that when one of the py.test harnesses finishes it pulls down the zookeeper (the ports are hardcoded) causing the tests running in the other harness to fail.

In many real integrations(like with a CI framework like jenkins):
1) The tests run fail because of the above stated reason.
2) We cannot change ports, which is needed since a lot of the ports used by other services running in a CI environment(P.S.: the offset is hardcoded in the class ZooKeeperCluster and so is the # of running ZK server instances)

Is there a reason for writing the harness the way it is?
